server/api/batch_tallies.py

```diff
@@ -1,34 +1,34 @@
 from datetime import datetime
 import uuid
-from sqlalchemy.orm.session import Session
 from flask import request, jsonify, Request
 from werkzeug.exceptions import BadRequest, NotFound, Conflict
 
 from . import api
 from ..database import db_session
 from ..models import *  # pylint: disable=wildcard-import
 from ..auth import restrict_access, UserType
-from ..util.process_file import (
-    process_file,
-    serialize_file,
-    serialize_file_processing,
+from ..worker.tasks import (
     UserError,
+    background_task,
+    create_background_task,
 )
+from ..util.file import serialize_file, serialize_file_processing
 from ..util.csv_download import csv_response
 from ..util.csv_parse import decode_csv_file, parse_csv, CSVValueType, CSVColumnType
 from ..activity_log.activity_log import UploadFile, activity_base, record_activity
 
 BATCH_NAME = "Batch Name"
 
 
-def process_batch_tallies_file(
-    session: Session, jurisdiction: Jurisdiction, file: File
-):
-    def process():
+@background_task
+def process_batch_tallies_file(jurisdiction_id: str):
+    jurisdiction = Jurisdiction.query.get(jurisdiction_id)
+
+    def process() -> None:
         # We only support one contest for batch audits, so we can just take the
         # first contest from the jurisdiction's universe.
-        assert len(jurisdiction.contests) == 1
-        contest = jurisdiction.contests[0]
+        assert len(list(jurisdiction.contests)) == 1
+        contest = list(jurisdiction.contests)[0]
```

Comment:
are we doing the query twice here? Should we store `list(jurisdiction.contest)` and then work off that?

Reply:
sqlalchemy caches relationship query results, so there's no performance hit, it's just for making the typechecker not complain